rest/bulk_api.go

```diff
@@ -397,21 +393,35 @@ func (h *handler) handleBulkGet() error {
 				err = base.HTTPErrorf(http.StatusBadRequest, "Invalid doc/rev ID in _bulk_get")
 			} else {
 				attsSince, err = db.GetStringArrayProperty(doc, "atts_since")
-				if revsLimit > 0 {
-					revsFrom, err = db.GetStringArrayProperty(doc, "revs_from")
-					if revsFrom == nil {
-						revsFrom = attsSince // revs_from defaults to same value as atts_since
+
+				if h.getBoolQuery("revs") {
+					// Try to pull out a per-doc revs limit, otherwise use request limit.
+					if raw, ok := doc["revs_limit"]; !ok {
+						docRevsLimit = int(h.getIntQuery("revs_limit", math.MaxInt32))
+					} else if val, ok := raw.(float64); ok && val >= 0 {
+						docRevsLimit = int(val)
+					} else {
+						err = base.HTTPErrorf(http.StatusBadRequest, "Invalid revs_limit for doc: %s in _bulk_get", docid)
```

Comment:
I found the logic here a bit hard to parse as if/else if/else, with multiple usages of ok.  I think it would be clearer with the second else nested, to make it clearer that it's associated with the case where revs_limit is present (but invalid).

Reply:
Agreed. Thanks for the review, I'll update today.